macaw-x86-syntax/README.md

```diff
@@ -0,0 +1,36 @@
+# macaw-x86-syntax
+
+This package provides concrete syntax for macaw-x86-symbolic types and
+operations.
+
+Concretely, it implements a `ParserHooks` for use with [`crucible-syntax`][syn].
+This `ParserHooks` supports the following types and operations:
+
+**Types**:
+
+- `X86Regs`: the struct of all x86_64 registers
```

Comment:
Should `X86Reg` also be defined here?

Reply:
Ah, this is kind of weird... `X86Regs` is actually something the programmer can write in an S-expression file. `X86Reg` isn't an object-language type, it's just something I'm using as notation to denote the set of syntactic things that can be passed to `get-reg` and `set-reg`... Technically, `{g,s}et-reg` are untypable in Crucible itself, they're kind of dependently-typed in the argument register (consider, e.g., if we had syntax for naming a non-64-bit register).

This weirdness is present in all the `*-syntax` READMEs... Really, they should probably all have Martin-Lof style judgments for the combination of syntactic well-formedness and well-typedness that the parser recognizes, perhaps alongside these more readable, "informal type signatures".

Do you have any thoughts on how we should tackle this? I could at least file an issue upstream in the Crucible repo. The README for `crucible-syntax` is pretty woefully incomplete...

All of this being said, this is almost all just developer-facing, so it's not clear to me how much effort we really want/need to spend precisely specifying these parser's behaviors.

Reply:
Ah, OK. No need to give precise typing rules here: I was just wondering if I had overlooked something. I am fine with being a little hand-wavey here.